concepts/functions/about.md

```diff
@@ -1,3 +1,93 @@
 # About
 
-TODO
+Functions are bodies of code containing one or more [statements or expressions][statements or expressions]. A function may optionally
+[return an value][return value].
+
+Rust style is to use [snake case][snake case] for a function name, which is prefaced by the `fn` keyword for a function definition. For example
+
+```rust
+fn do_nothing() {}
+```
+
+The parentheses surround zero or more parameters, separated by commas. A parameter is a binding to a value of a particular type which is passed into the function.
+Even if there are no parameters the parentheses are still required in the function definition. The actual value passed to a parameter is called
+an argument.
+
+The combination of parameters and return value is known as the function's signature. A function's signature requires that each parameter must
+have its type annotated.
+
+The curly braces enclose the body of the function: its statements and expressions. Even if the function contains no statements or expressions the
+curly braces are still required for the function definition.
+
+In the following example the function takes in one parameter of type `i32`, binds it to the name `value`, and prints it.
+
+```rust
+fn print_integer(value: i32) {
+    println!("{:?}", value);
+}
+```
+
+Note the parameter's definition. Each parameter is defined in the format `name: Type`.
+
+A function can also return a value. By default, the output of the final expression is returned.
+In the following example the function has one `i32` parameter and returns its double.
+
+```rust
+fn double_integer(value: i32) -> i32 {
+    value * 2
+}
+```
+
+The `-> i32` indicates that the function returns an `i32`. Unlike parameters, the returned value is not named.
+
+It is possible to exit from a function early with the `return` keyword, like so:
+
+```rust
+fn long_function() -> i32 {
+    let some_condition = false;
+    // code snipped
+    if (some_condition) {
+        return 0;
+    }
+    // more code snipped
+    42
+}
+```
+
+`const fn` is used to define a [constant function][constant function], which can be evaluated at compile time. Maing a function constant
+is a way of declaring that the function won't change in ways that are invalid for a `const fn`. Since this may prevent future optimizations,
```

Comment:
> Maing a function constant is a way of declaring that the function won't change in ways that are invalid for a `const fn`.

- Typo: "Making" not "Maing"
- This sentence doesn't tell me anything. I recommend removing it.

Reply:
I'm sorry, which sentence? The one with the typo, or the one starting with "Since"?

Reply:
The one I quoted, starting with the typo.